src/App.tsx

```diff
@@ -26,56 +26,101 @@ type ReorderOptions = {
   isReversed: boolean;
 };
 
-// Use this function in the render method to prepare goods
 export function getReorderedGoods(
   goods: string[],
   { sortType, isReversed }: ReorderOptions,
 ) {
-  // To avoid the original array mutation
   const visibleGoods = [...goods];
 
-  // Sort and reverse goods if needed
-  // eslint-disable-next-line no-console
-  console.log(sortType, isReversed);
+  switch (sortType) {
+    case SortType.ALPHABET:
+      visibleGoods.sort((a, b) => a.localeCompare(b));
+      break;
+    case SortType.LENGTH:
+      visibleGoods.sort((a, b) => a.length - b.length);
+      break;
+    default:
+  }
+
+  if (isReversed) {
+    visibleGoods.reverse();
+  }
 
   return visibleGoods;
 }
 
-// DON'T save goods to the state
-// type State = {
-//   isReversed: boolean,
-//   sortType: SortType,
-// };
+type State = {
+  isReversed: boolean;
+  sortType: SortType;
+};
 
 export const App: React.FC = () => {
+  const [sortOptions, setSortOptions] = useState<State>({
+    isReversed: false,
+    sortType: SortType.NONE,
+  });
+  const { isReversed, sortType } = sortOptions;
+
+  const handleClickSortType = (newType: SortType) => {
+    setSortOptions(prev => ({ ...prev, sortType: newType }));
+  };
+
+  const handleClickReverse = () => {
+    setSortOptions(prev => ({ ...prev, isReversed: !prev.isReversed }));
+  };
+
+  const handleClickSortDefault = () => {
+    setSortOptions({
+      isReversed: false,
+      sortType: SortType.NONE,
+    });
+  };
+
   return (
     <div className="section content">
       <div className="buttons">
-        <button type="button" className="button is-info is-light">
+        <button
+          type="button"
+          onClick={() => handleClickSortType(SortType.ALPHABET)}
+          className={`button is-info ${sortType !== SortType.ALPHABET && 'is-light'}`}
+        >
           Sort alphabetically
         </button>
 
-        <button type="button" className="button is-success is-light">
+        <button
+          type="button"
+          onClick={() => handleClickSortType(SortType.LENGTH)}
+          className={`button is-success ${sortType !== SortType.LENGTH && 'is-light'}`}
+        >
           Sort by length
         </button>
 
-        <button type="button" className="button is-warning is-light">
+        <button
+          type="button"
+          onClick={handleClickReverse}
+          className={`button is-warning ${!isReversed && 'is-light'}`}
```

Comment:
Again, ensure that the conditional expression in the template literal returns a valid string. Use a ternary operator to handle the class name correctly.